client/state/jetpack/credentials/actions.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * Internal dependencies
+ */
+import {
+	JETPACK_CREDENTIALS_RECEIVE,
+	JETPACK_CREDENTIALS_REQUEST,
+	JETPACK_CREDENTIALS_REQUEST_FAILURE,
+	JETPACK_CREDENTIALS_REQUEST_SUCCESS,
+	JETPACK_CREDENTIALS_UPDATE,
+	JETPACK_CREDENTIALS_UPDATE_FAILURE,
+	JETPACK_CREDENTIALS_UPDATE_SUCCESS
+} from 'state/action-types';
+import wp from 'lib/wp';
+
+/**
+ * Fetch the Jetpack credentials for a certain site.
+ *
+ * @param  {Int}      siteId      ID of the site.
+ * @return {Function}             Action thunk to fetch the Jetpack credentials when called.
+ */
+export const fetchCredentials = ( siteId ) => {
+	return ( dispatch ) => {
+		dispatch( {
+			type: JETPACK_CREDENTIALS_REQUEST,
+			siteId
+		} );
+
+		return wp.undocumented().fetchJetpackCredentials( siteId )
+			.then( ( response ) => {
+				const credentials = response.credentials || {};
+				dispatch( {
+					type: JETPACK_CREDENTIALS_RECEIVE,
+					siteId,
+					credentials
+				} );
+				dispatch( {
+					type: JETPACK_CREDENTIALS_REQUEST_SUCCESS,
+					siteId
+				} );
+			} ).catch( error => {
+				dispatch( {
+					type: JETPACK_CREDENTIALS_REQUEST_FAILURE,
+					siteId,
+					error: error.message
+				} );
+			} );
+	};
```

Comment:
Just for fun I'm going to scratch together approximately what this would look like if it were written in the data layer…

```js
// in `state/data-layer/wpcom-http/activity-log/get-credentials/index.js`

const fetchCredentials = ( { dispatch }, { siteId } ) => dispatch( http( {
	method: 'GET',
	path: `/activity-log/${ siteId }/get-credentials`
} ) );

const storeCredentials = ( { dispatch }, { siteId }, creds ) => dispatch( {
	type: CREDS_STORE,
	siteId,
	creds
} );

const announceFailure = ( { dispatch } ) => dispatch( errorNotice(
	i18n.translate( 'Unexpected problem retrieving the credentials for your JP site, please try again.' )
) );

export default {
	[ JP_CRED_REQUEST ]: [ dispatchRequest( fetchCreds, storeCreds, announceFailure ) ]
}
```